### PR TITLE
Deployment to Heroku, Model Files, Final Report

### LINKED ISSUE #2: Deploy the Web Application to Heroku
Use [Heroku](https://www.heroku.com/) to deploy the web application (currently : Streamlit & Future Milestone : Flask) to the web.

### LINKED ISSUE #8: Enhanced User Interface
Since Streamlit provides little tools to specialize the Application, Improve the UI with a customized look & design (preferably in Flask).

### PR DESCRIPTION
**Pull request by Dineshkumar536**
# Description
This project analyse sentiments,Keywords,Polarity,Subjectivity and Summary of given text using NLTK features and Naive Bayes model.

Fixes # [ISSUE]

# Type of Change:
**Delete irrelevant options.**

- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.

# Checklist:
**Delete irrelevant options.**

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by khanfarhan10** [comment]
Is this the final model that is deployed on Heroku?

File: code_notebook(samples_images).md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+## NOTEBOOK (CODE IMAGES)
```
Comment (khanfarhan10) [inline code_notebook(samples_images).md]: What does this file do exactly?

File: models_files/preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#import all necessary packages
+import nltk
+nltk.download('punkt')
+from nltk.stem import PorterStemmer
+from nltk.tokenize import word_tokenize
+from nltk.corpus import stopwords
+nltk.download('stopwords')
```
Comment (khanfarhan10) [inline models_files/preprocessing.py]: Maybe push the downloaded softwares beforehand?

**Review by khanfarhan10 — CHANGES_REQUESTED** [review]
Resolve Issues in the Code. Please complete it within today as after that PRs won't be entertained for Winter of Code.

**Comment by khanfarhan10** [comment]
Rename `app.py` to `FlaskApp.py`  and commit. Also resolve the problems mentioned above. What are code images?? Are they for the blog/Final Report???

Also you can download Natural Language Toolkit Stopwords and include them in a folder instead of someone having to download them from the official website. Try to keep the Project as Standalone as possible.

You can always use
<b>`</b> - This character (Single Quotes) for inline code.

This is an example of `inline code`.

To write your big code : Use Triple Quotes from the same character (<b>```</b>).

For example:

```
Your Code Here
for i in range(10):
    print("Merry Christmas !")
```

**Comment by Dineshkumar536** [comment]
Code images are for final report these are images of my jupyter notebook

**Comment by khanfarhan10** [comment]
> Code images are for final report these are images of my jupyter notebook


Please don't upload so many images as they will have no use and will further increase Repository Size unnecessarily.

Try converting your Jupyter Notebook to Markdown File instead. Use [this colab](https://colab.research.google.com/drive/1NYOHMC-HzTuvd5GYqOfI1rSiRd0sVRhn?usp=sharing) file for the same.

Alternatively use this command : `! jupyter nbconvert --to markdown MNIST_TSNE_KNN.ipynb`


Also you can use this extension on VSCode for viewing your md file : [Markdown Preview Enhanced](https://marketplace.visualstudio.com/items?itemName=shd101wyy.markdown-preview-enhanced)

**Comment by Dineshkumar536** [comment]
So may i delete that images file?

File: code_notebook(samples_images).md (hunk 1 above)
Comment (Dineshkumar536) [reply code_notebook(samples_images).md]: i removed this and added jupyter notebook to markdown file

Comment (Dineshkumar536) [reply code_notebook(samples_images).md]: https://sentimentproject.herokuapp.com/
this is link of deployed model

File: models_files/preprocessing.py (hunk 2 above)
Comment (Dineshkumar536) [reply models_files/preprocessing.py]: okay

**Comment by khanfarhan10** [comment]
I cannot follow what you've currently done. You just converted the whole notebook to a markdown file which I don't think is what you were trying to accomplish.

Delete that markdown file as it serves no purpose.

 I had objections against using code images only.
 
 Also there is a folder with your name. Please note that only your final report will contain your name. Change as required.

**Comment by khanfarhan10** [comment]
Also, I cannot see any changes in the Description section of [this pull request](https://github.com/khanfarhan10/TextSentimentAnalysis/pull/29)

**Comment by Dineshkumar536** [comment]
I have changed the name of pull request

File: code_notebook(samples_images).md (hunk 1 above)
Comment (khanfarhan10) [reply code_notebook(samples_images).md]: No need to convert the ipynb to md, you can directly commit the notebook and delete all the code images

**Comment by Dineshkumar536** [comment]
I have deleted all code images

File: code_notebook(samples_images).md (hunk 1 above)
Comment (khanfarhan10) [reply code_notebook(samples_images).md]: Also you need to change the deployed app on heroku and give everyone the credits for it including WinterofCode, @paltannistha , @khanfarhan10 , @Vamsi995 , @Ankitasareen and yourself @Dineshkumar536 . Also you need to redeploy it as your fellowmates are regularly making changes to the repository.

However, you can keep the developer credits at the bottom of the page as you have made the app yourself.

**Comment by khanfarhan10** [comment]
> I have deleted all code images

I don't see any changes you have to push changes for it to be visible.

**Comment by Dineshkumar536** [comment]
Means i have to add names of my all mates and mentor in my web page right?

**Comment by Dineshkumar536** [comment]
Ok, I will check.

**Comment by khanfarhan10** [comment]
> Means i have to add names of my all mates and mentor in my web page right?
 Yes.

Also you have to fix the webapp. There appears to be a bug there.

**Comment by Dineshkumar536** [comment]
https://sentimentproject.herokuapp.com/
Try this

**Comment by khanfarhan10** [comment]
> https://sentimentproject.herokuapp.com/ Try this

It still shows `500 Internal Server Error`

**Comment by khanfarhan10** [comment]
> I have changed the name of pull request


Description still missing @Dineshkumar536 !

**Comment by Dineshkumar536** [comment]
But it is working correctly in my device and other also

**Comment by Dineshkumar536** [comment]
Where to give description?

**Comment by khanfarhan10** [comment]
> Where to give description?

Can't you see? :((( It's on the top of this PR. View it on GitHub and not from Gmail. I'm tired of repeating this multiple times for you.
I'm not going to do it again for you.

**Comment by Dineshkumar536** [comment]
Okay understood👍

**Comment by khanfarhan10** [comment]
Done creating changes.

**Pull request merged**